src/genericComponents/Button.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import styled from 'styled-components';
+
+const ButtonPrimary = (props: any) => {
+  const { onClick, text, variant } = props;
+
+  // className={`${getVariantClassName(variant)}${className ? ' ' + className : ''}`}
+
+  return (
+    <CAGButton
+      className={`CAGButton.${variant}-class`}
```

Comment:
You should just do `${variant}-class` - no need to have `CAGButton` because the class will already belong only to CAGButton based on the scope of your styled components block below